### DISCUSSION TITLE
Use Request in Fastapi to get post json data request, it throw a error : exception in asgi application

### DISCUSSION THREAD

**Discussion by cloudy-Meng** [fastapi/fastapi#12498] [Questions] [answered]
### Privileged issue

- [X] I'm @tiangolo or he asked me directly to create an issue here.

### Issue Content

code:
![image](https://github.com/user-attachments/assets/57cba8ee-0b23-41b9-bc65-5061680dca19)
error:
![image](https://github.com/user-attachments/assets/68b08ec8-d54a-40dc-9d69-e75b19862cde)

**Comment by YuriiMotov** [comment, marked as answer]
Please, next time add the code example and error message as a text, not picture.

To fix the error you just need to make your endpoint function async and add `await` before `request.json()`:
```
@app.post("/")
async def index(request: Request):
    return {"request": await request.json()}
```
